From 8d9c489939c7258989d80297efb44761eb5e0160 Mon Sep 17 00:00:00 2001
From: Yvan Volochine <yvan.volochine@gmail.com>
Date: Mon, 22 Jul 2013 13:25:15 +0200
Subject: [PATCH] scide: multi-editor: change jump-to-tab shortcuts key on Mac
 and Windows

- Mac uses "Cmd+Number"
- Windows uses "Ctrl+Number"
- Linux uses "Alt+Number"
---
 editors/sc-ide/widgets/multi_editor.cpp | 27 ++++++++++++++++++---------
 1 file changed, 18 insertions(+), 9 deletions(-)

diff --git a/editors/sc-ide/widgets/multi_editor.cpp b/editors/sc-ide/widgets/multi_editor.cpp
index 67c1601..d7a52ae 100644
--- a/editors/sc-ide/widgets/multi_editor.cpp
+++ b/editors/sc-ide/widgets/multi_editor.cpp
@@ -239,40 +239,49 @@ class DocumentSelectPopUp : public QDialog
     // A signal mapper to avoid creating custom slots for each tab
     QSignalMapper *mapper = new QSignalMapper(this);
 
+    QString keySeq;
+    #if defined(Q_WS_MAC)
+        keySeq = "Cmd+";
+    #elif defined(Q_WS_WIN)
+        keySeq = "Ctrl+";
+    #else
+        keySeq = "Alt+";
+    #endif
+
     // Shortcuts for tabs (1 to 8)
-    QShortcut *s1 = new QShortcut(QKeySequence("Alt+1"), this);
+    QShortcut *s1 = new QShortcut(QKeySequence(keySeq + "1"), this);
     connect( s1, SIGNAL(activated()), mapper, SLOT(map()) );
     mapper->setMapping(s1, 0);
 
-    QShortcut *s2 = new QShortcut(QKeySequence("Alt+2"), this);
+    QShortcut *s2 = new QShortcut(QKeySequence(keySeq + "2"), this);
     connect( s2, SIGNAL(activated()), mapper, SLOT(map()) );
     mapper->setMapping(s2, 1);
 
-    QShortcut *s3 = new QShortcut(QKeySequence("Alt+3"), this);
+    QShortcut *s3 = new QShortcut(QKeySequence(keySeq + "3"), this);
     connect( s3, SIGNAL(activated()), mapper, SLOT(map()) );
     mapper->setMapping(s3, 2);
 
-    QShortcut *s4 = new QShortcut(QKeySequence("Alt+4"), this);
+    QShortcut *s4 = new QShortcut(QKeySequence(keySeq + "4"), this);
     connect( s4, SIGNAL(activated()), mapper, SLOT(map()) );
     mapper->setMapping(s4, 3);
 
-    QShortcut *s5 = new QShortcut(QKeySequence("Alt+5"), this);
+    QShortcut *s5 = new QShortcut(QKeySequence(keySeq + "5"), this);
     connect( s5, SIGNAL(activated()), mapper, SLOT(map()) );
     mapper->setMapping(s5, 4);
 
-    QShortcut *s6 = new QShortcut(QKeySequence("Alt+6"), this);
+    QShortcut *s6 = new QShortcut(QKeySequence(keySeq + "6"), this);
     connect( s6, SIGNAL(activated()), mapper, SLOT(map()) );
     mapper->setMapping(s6, 5);
 
-    QShortcut *s7 = new QShortcut(QKeySequence("Alt+7"), this);
+    QShortcut *s7 = new QShortcut(QKeySequence(keySeq + "7"), this);
     connect( s7, SIGNAL(activated()), mapper, SLOT(map()) );
     mapper->setMapping(s7, 6);
 
-    QShortcut *s8 = new QShortcut(QKeySequence("Alt+8"), this);
+    QShortcut *s8 = new QShortcut(QKeySequence(keySeq + "8"), this);
     connect( s8, SIGNAL(activated()), mapper, SLOT(map()) );
     mapper->setMapping(s8, 7);
     // Last tab
-    QShortcut *s9 = new QShortcut(QKeySequence("Alt+9"), this);
+    QShortcut *s9 = new QShortcut(QKeySequence(keySeq + "9"), this);
     connect( s9, SIGNAL(activated()), this, SLOT(showLastDocument()) );
 
     connect( mapper, SIGNAL(mapped(int)), mTabs, SLOT(setCurrentIndex(int)) );
-- 
1.8.4

